src/main/java/mate/academy/AsyncRequestProcessor.java

```diff
@@ -1,16 +1,23 @@
 package mate.academy;
 
+import java.util.Map;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.Executor;
 
 public class AsyncRequestProcessor {
+    private final Map<String, UserData> cache = new ConcurrentHashMap<>();
     private final Executor executor;
 
     public AsyncRequestProcessor(Executor executor) {
         this.executor = executor;
     }
 
     public CompletableFuture<UserData> processRequest(String userId) {
-        return null;
+        var data = new UserData(userId, "Details for " + userId);
+        return CompletableFuture.supplyAsync(() -> {
+            cache.computeIfAbsent(userId, userDetails -> data);
+            return data;
+        }, executor);
```

Comment:
* wrong naming of the function param inside `computeIfAbsent()` you pass id there, not details
* I do not like how we build the `data` object before cache is checked anyway
* why don't you return the result of `computeIfAbsent()` call? please [read more](https://www.baeldung.com/java-map-computeifabsent) on how it works
* we should sleep the current thread when we generate the `data` object to simulate some DB access time, let's add a `UserData getUserDataById(userId)` method for that 
```suggestion
        return CompletableFuture.supplyAsync(() -> cache.computeIfAbsent(userId, this::getUserDataById), executor);
```